wormhole-connect/src/hooks/useFetchTransactionHistory.ts

```diff
@@ -0,0 +1,243 @@
+    if (!fromChain) {
+      return;
+    }
+
+    const tokenConfig = getTokenById({
+      chain: fromChain,
+      address: standarizedProperties.tokenAddress,
+    });
+
+    // Skip if we don't have the source token
+    if (!tokenConfig) {
+      return;
+    }
+
+    const toChain = chainIdToChain(toChainId) as Chain;
+    const toChainConfig = config.chains[toChain];
+
+    const receivedTokenKey =
+      tokenConfig.nativeChain === toChain
+        ? tokenConfig.key
+        : getWrappedToken(tokenConfig)?.key;
+
+    // Reverse engineering the conversion rate:
+    //   data.tokenAmount is the actual value and standarizedProperties.amount is the denomination.
+    //   There has been unexpected conversaion rates in API responses. Therefore we are
+    //   sniffing out the decimals from the conversion rate of between these two values.
+    const conversionRate =
+      Number(standarizedProperties.amount) / Number(data.tokenAmount);
+    const decimals = conversionRate.toString().length - 1;
+
+    const sentAmountDisplay = sdkAmount.display(
+      {
+        amount: standarizedProperties.amount,
+        decimals,
+      },
+      0,
+    );
+
+    const receiveAmountValue = Math.max(
+      standarizedProperties.amount - standarizedProperties.fee,
+      0,
+    );
+    const receiveAmountDisplay = sdkAmount.display(
+      {
+        amount: receiveAmountValue.toString(),
+        decimals,
+      },
+      0,
+    );
+
+    const feeAmountDisplay = sdkAmount.display(
+      {
+        amount: standarizedProperties.fee,
+        decimals,
+      },
+      0,
+    );
+
+    const txData: Transaction = {
+      txHash: sourceChain.transaction?.txHash,
+      sender: standarizedProperties.fromAddress || sourceChain.from,
+      amount: sentAmountDisplay,
+      amountUsd: data.usdAmount,
+      recipient: standarizedProperties.toAddress,
+      toChain,
+      fromChain,
+      tokenKey: tokenConfig.key,
+      receivedTokenKey,
+      receiveAmount: receiveAmountDisplay,
+      relayerFee: {
+        fee: Number(feeAmountDisplay),
+        tokenKey: toChainConfig?.gasToken || '',
+      },
+      tokenAddress: standarizedProperties.tokenAddress,
+      senderTimestamp: sourceChain?.timestamp,
+      receiverTimestamp: targetChain?.timestamp,
+    };
+
+    return txData;
+  };
+
+  const SUPPORTED_APPIDS = {
+    PORTAL_TOKEN_BRIDGE: parseTokenBridgeTx,
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const parseTransactions = useCallback((allTxs: Array<any>) => {
+    const parsedTxs: Array<Transaction> = allTxs.map((tx) => {
+      const appIds = tx.content?.standarizedProperties?.appIds || [];
+
+      let txParser;
+
+      appIds.forEach((appId) => {
+        if (typeof SUPPORTED_APPIDS[appId] === 'function') {
+          txParser = SUPPORTED_APPIDS[appId];
+        }
+      });
+
+      return txParser?.(tx);
+    });
+
+    return parsedTxs.filter((tx) => !!tx);
```

Comment:
Feel free to take this or leave it, but you could make this logic easier to read and run a little faster

```suggestion
    return allTxs.map((tx) => {
      const appIds = tx.content?.standarizedProperties?.appIds || [];
      for (const appId of appIds) {
        const parser = SUPPORTED_APPIDS[appId];
        if (parser) return parser(tx);
      }
    }).filter((tx) => !!tx);
```

Reply:
I'll make this more simple 👍